components/create/editor/Publishing.tsx

```diff
@@ -0,0 +1,317 @@
+          type: "Info",
+          lifetime: 60,
+        });
+
+        const result = await sdk.model.markets.create(params);
+        const marketId = result.saturate().unwrap().market.marketId;
+
+        editor.published(marketId);
+
+        notifications.pushNotification(
+          "Transaction successful! Awaiting indexer.",
+          {
+            autoRemove: true,
+            type: "Info",
+            lifetime: 20,
+          },
+        );
+
+        const indexedStatus = await poll(
+          async () => {
+            return checkMarketExists(sdk.indexer.client, marketId);
+          },
+          {
+            intervall: 1000,
+            timeout: 6 * 1000,
+          },
+        );
+
+        if (indexedStatus === PollingTimeout) {
+          router.push(`/markets/await/${marketId}`);
+        } else {
+          notifications.pushNotification(
+            "Market has been created and indexed! Redirecting to market page.",
+            {
+              autoRemove: true,
+              type: "Success",
+              lifetime: 15,
+            },
+          );
+
+          router.push(`/markets/${marketId}`);
+        }
+
+        setTimeout(() => {
+          editor.reset();
+        }, 2000);
+      } catch (error) {
+        let type: NotificationType = "Error";
+        let errorMessage = "Unknown error occurred.";
+
+        if (StorageError.is(error)) {
+          errorMessage = "IPFS metadata upload failed.";
+        }
+
+        if (isArray(error?.docs)) {
+          errorMessage = error.docs[0];
+        }
+
+        if (error?.message === "Cancelled") {
+          type = "Info";
+          errorMessage = "Transaction cancelled";
+        }
+
+        notifications.pushNotification(errorMessage, {
+          autoRemove: true,
+          type: type,
+          lifetime: 15,
+        });
+
+        console.error(error);
+      }
+
+      setIsTransacting(false);
+    }
+  };
+
+  const baseAssetLiquidityRow = editor.form?.liquidity?.rows.find(
+    (row) => row.asset === editor.form.currency,
+  );
+
+  const bondCost =
+    editor.form.moderation === "Permissionless"
+      ? constants?.markets.validityBond
+      : constants?.markets.advisoryBond;
+
+  const oracleBond = constants?.markets.oracleBond;
+
+  const ztgCost = new Decimal(bondCost ?? 0)
+    .plus(oracleBond ?? 0)
+    .plus(
+      editor.form.liquidity?.deploy && editor.form.currency === "ZTG"
+        ? new Decimal(baseAssetLiquidityRow?.value).mul(2).toNumber()
+        : 0,
+    );
+
+  const foreignCurrencyCost =
+    editor.form.liquidity?.deploy && editor.form.currency !== "ZTG"
+      ? new Decimal(baseAssetLiquidityRow?.value ?? 0).mul(2).toNumber()
+      : null;
+
+  return (
+    <>
+      <div className="">
+        <div className="inline-block mb-6 center">
+          <div className="mb-2 center w-full">
+            <div className="relative">
+              <div className="relative">
+                <TransactionButton
```

Comment:
We should check that the user has enough balance.

If I remove liquidity by putting advised, this modal doesn't update:
<img width="744" alt="image" src="https://github.com/zeitgeistpm/ui/assets/4950844/6d68f73b-4bfb-443a-92ad-7e77a7804791">

Reply:
Coming in the next build